### PR TITLE
bin: add rocksdb log-dir option

### LINKED ISSUE #1360: feature request: configuration about rocksdb log file
Currently rocksdb log is stored in ``data/db/LOG`` by default. 

Better add a configuration entry:

``--rocksdb-log-file``

All data & log should be stored separately, for best practise.

### PR DESCRIPTION
**Pull request by overvenus**
Add rocksdb log-dir option.

Pending ngaut/rust-rocksdb#61

PTAL @siddontang @BusyJay @zhangjinpeng1987 

Close #1360

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -111,6 +111,13 @@ max-manifest-file-size = "20MB"
 # If true, the database will be created if it is missing.
 create-if-missing = true
 
+# This specifies the info LOG dir.
+# If it is empty, the log files will be in the same dir as data.
+# If it is non empty, the log files will be in the specified dir,
+# and the db data dir's absolute path will be used as the log file
```
Comment (siddontang) [inline etc/config-template.toml]: > the db data dir's absolute path will be used as the log file name's prefix

What's the prefix here? Seem that we still use `LOG` directly.

Comment (overvenus) [inline etc/config-template.toml]: Let's say `log-dir = "/path/to/dir"`, then the actual LOG file is named `path_to_dir_LOG`, and it is stored under `/path/to/dir`.

Comment (siddontang) [inline etc/config-template.toml]: The actual name looks strange...

**Comment by siddontang** [comment]
Please update cargo rocksdb

**Comment by siddontang** [comment]
LGTM
PTAL @BusyJay @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
LGTM

File: etc/config-template.toml [hunk 2]
```diff
@@ -116,6 +116,13 @@ max-manifest-file-size = "20MB"
 # If true, the database will be created if it is missing.
 create-if-missing = true
 
+# This specifies the info LOG dir.
```
Comment (BusyJay) [inline etc/config-template.toml]: What does `info` mean here?

File: etc/config-template.toml [hunk 3]
```diff
@@ -116,6 +116,13 @@ max-manifest-file-size = "20MB"
...
+# If it is empty, the log files will be in the same dir as data.
```
Comment (BusyJay) [inline etc/config-template.toml]: What data?

File: etc/config-template.toml [hunk 4]
```diff
@@ -116,6 +116,13 @@ max-manifest-file-size = "20MB"
...
+# If it is non empty, the log files will be in the specified dir,
```
Comment (BusyJay) [inline etc/config-template.toml]: You mean not?

File: etc/config-template.toml [hunk 5]
```diff
@@ -116,6 +116,13 @@ max-manifest-file-size = "20MB"
...
+# and the db data dir's absolute path will be used as the log file
+# name's prefix.
+# log-dir = "/path/to/log-dir"
```
Comment (BusyJay) [inline etc/config-template.toml]: Will all TiKV's log be put in this directory too?

Comment (overvenus) [inline etc/config-template.toml]: Just a **rocksdb's LOG**, not the log specified by `-f` argument.

**Comment by BusyJay** [comment]
I think it's better to keep the rocksdb's log and TiKV's log together by default. @siddontang

**Comment by siddontang** [comment]
Yes, I said before in https://github.com/pingcap/tikv/issues/1360, we can use the directory of `--log-file`.

**Comment by siddontang** [comment]
or we can use `log-dir` to replace `log-file`, but this is not backward compatibility.

File: src/bin/tikv-server.rs [hunk 6]
```diff
@@ -226,6 +226,22 @@ fn get_rocksdb_db_option(config: &toml::Value) -> RocksdbOptions {
     let create_if_missing = get_toml_boolean(config, "rocksdb.create-if-missing", Some(true));
     opts.create_if_missing(create_if_missing);
 
+    let mut log_dir = get_toml_string(config, "rocksdb.log-dir", Some(String::new()));
```
Comment (siddontang) [inline src/bin/tikv-server.rs]: I think we can remove log-dir now.
/cc @BusyJay

**Comment by siddontang** [comment]
LGTM

PTAL @BusyJay @zhangjinpeng1987

**Comment by BusyJay** [comment]
LGTM, I propose to disable all command line arguments so that tikv-ctl can reuse the same settings as tikv-server.

**Pull request merged**